build-system/common/git.js

```diff
@@ -207,8 +249,8 @@ function gitCommitFormattedTime(ref = 'HEAD') {
  * a local workspace.
  * @return {string}
  */
-function gitMergeBaseLocalMaster() {
-  return getStdout('git merge-base master HEAD').trim();
+function gitMergeBaseMaster() {
+  return getStdout(`git merge-base ${gitFindMaster()} HEAD`).trim();
```

Comment:
This won't work either, because `git merge-base` also takes a commit or a local branch name.

![image](https://user-images.githubusercontent.com/26553114/80535974-a58bd000-896f-11ea-9f9e-78606c9ab2b0.png)

Reply:
Could I see the output of
```
$ git remote get-url upstream

$ git ls-remote --heads upstream | grep refs/heads/master
```

Reply:
![image](https://user-images.githubusercontent.com/26553114/80537582-4b403e80-8972-11ea-85cd-49150c3ea316.png)

Reply:
I'm very surprised. Thanks for verifying. I guess two steps are required for broader compatibility, then:
```
$ git rev-parse upstream/master
$ git cherry <hash>
```
and similarly for `merge-base`.

I'm sorry `git` is so gimped for you! I rely on remote branch management all the time!

Reply:
Just to reiterate how strange this is, see git's own docs:

https://git-scm.com/docs/git-merge-base#_discussion_on_fork_point_mode
> In such a case, `git merge-base origin/master topic` would return ...

https://git-scm.com/docs/git-cherry
> `$ git cherry origin/master topic`

Reply:
This issue should be resolved in the latest update. It's not something I could reproduce, but I'm pretty confident the method is broadly compatible. I also added handling for the case where a user has an upstream remote, but has never fetched it, in which case the local `master` branch will be used for `merge-base` and `cherry`.